tests/unit/task_system_test.c

```diff
@@ -478,18 +477,16 @@ create_task_system_without_bg_threads(void *datap)
    struct CTEST_IMPL_DATA_SNAME(task_system) *data =
       (struct CTEST_IMPL_DATA_SNAME(task_system) *)datap;
 
-   // no background threads by default.
-   ZERO_ARRAY(data->num_bg_threads);
-
    platform_status rc = STATUS_OK;
 
-   // Background threads are OFF by default.
-   rc = task_system_create(data->hid,
-                           data->ioh,
-                           &data->tasks,
-                           TRUE, // Use statistics,
-                           data->num_bg_threads,
-                           trunk_get_scratch_size());
+   // no background threads by default.
+   uint64 num_bg_threads[NUM_TASK_TYPES] = {0};
+   rc = task_system_config_init(&data->task_cfg,
+                                TRUE, // use stats
+                                num_bg_threads,
+                                trunk_get_scratch_size());
+   platform_assert(SUCCESS(rc));
```

Comment:
It may be better to change this to `ASSERT_TRUE()` so that if this `rc` is not SUCCESS, the test will fail, rather than asserting.

(I have tried to only use `platform_assert()` to check return status from stuff not directly related to this test itself; e.g. on L 127 in this file.)